scripts/test.sh

```diff
@@ -1,35 +1,185 @@
+    --gtest_break_on_failure        Stop running tests if a test fails.
+
+EXAMPLES:
+    - Run unit tests and integration tests and measure test coverage.
+      The build directory is set by an environment variable called 'BUILD_DIR'
+      or will be set to 'opencbdc-tx/build' by default.
+    $ ./test.sh
+
+    - Run integration tests but do not run unit tests.  Do not measure
+      test coverage. The build directory is set by an environment variable
+      called 'BUILD_DIR' or will be set to 'opencbdc-tx/build' by default.
+    $ ./test.sh -nu -nc
+
+    - Run unit tests and integration tests and measure test coverage.  Set the
+      build directory to 'mybuild'.
+    $ ./test.sh -d mybuild
+
+    - Run only the ArchiverTest tests and measure test coverage.
+    $ ./test.sh --gtest_filter=ArchiverTest*
+
+    - Run only the ArchiverTest.client test.  Repeat it 4 times.
+      Don't measure test coverage.
+    $ ./test.sh --gtest_filter=ArchiverTest.client --gtest_repeat=4 -nc
+
+    - Run only unit tests and don't measure test coverage.
+      Stop the tests if any test fails.
+    $ ./test.sh -ni -nc --gtest_break_on_failure
+	"
+}
+
+# Parse command-line arguments.
+RUN_UNIT_TESTS="true"
+RUN_INTEGRATION_TESTS="true"
+MEASURE_COVERAGE="true"
+GTEST_FLAGS=()
+while [[ $# -gt 0 ]]
+do
+    case $1 in
+        -h|--help)
+            usage
+            exit 0
+            ;;
+        -d|--build-dir)
+            shift
+            ARG="$1"
+            if [[ $ARG == "" || ${ARG:0:1} == "-" ]]
+            then
+                echo -n "ERROR:  The -d flag was used, "
+                echo "but a valid build folder was not given."
+                exit 1
+            fi
+            BUILD_DIR=$ARG
+            shift
+            ;;
+        -ni|--no-integration-tests)
+            RUN_INTEGRATION_TESTS="false"
+            shift
+            ;;
+        -nu|--no-unit-tests)
+            RUN_UNIT_TESTS="false"
+            shift
+            ;;
+        -nc|--no-coverage)
+            MEASURE_COVERAGE="false"
+            shift
+            ;;
+        --gtest_*)
+            GTEST_FLAGS+=("$1")
+            shift
+            ;;
+        -*)
+            echo "ERROR:  unknown command-line option '${1}'."
+            exit 1
+            ;;
+        *)
+            echo "ERROR:  unknown command-line option '${1}'."
+            exit 1
+            ;;
+    esac
+done
+
+# If the build folder is not provided as a command-line argument or as an
+# environment variable, assume it's named 'build' and is located in the
+# top-level directory of the repo.  By defining the top-level directory relative
+# to the location of this script, the user can run this script from any folder.
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+REPO_TOP_DIR="${SCRIPT_DIR}/.."
+if [ -z ${BUILD_DIR+x} ]
+then
+    BUILD_DIR="${REPO_TOP_DIR}/build"
+fi
+
+if [ ! -d "$BUILD_DIR" ]
```

Comment:
We could create the build folder as you suggest, but then all the tests will crash or fail anyways since the new folder won't contain any code to test.  In that case, I was thinking it would be clearer for users to create the build folder themselves through whatever step they're taking to build the code (maybe by running build.sh).

However, I'm happy to do it either way!